File: jlogkim/Season2/038.py
def solution(s):
    answer = -1

    st = []
    
    for ch in s:
        if len(st)>0 and st[-1]== ch:
            st.pop()
        else:
            st.append(ch)

    if len(st)==0:
        return 1
    else:
        return 0

File: joonas/038.cpp
#include <string>
#include <stack>
using namespace std;

int solution(string s) {
    stack<char> x;
    x.push(s[0]);
    for (int i = 1; i < s.length(); ++i) {
        if(!x.empty() && x.top() == s[i]) x.pop();
        else x.push(s[i]);
    }
    return x.empty();
}